pkgs/games/katawa-shoujo/default.nix

```
@@ -0,0 +1,179 @@
+, libjpeg
+, libpng12
+, libX11
+, libXext
+, libXi
+, libXmu
+, runtimeShell
+, SDL
+, withSDL2 ? true
+, SDL_compat
+, SDL_image
+, SDL_ttf
+, undmg
+, zlib
+}:
+
+let
+  stdenv = stdenvNoCC;
+  srcDetails = rec {
+    x86_64-linux = {
+      urlSuffix = "%5blinux-x86%5d%5b18161880%5d.tar.bz2";
+      hash = "sha256-7FoFz88dWYHs2/pxkEwnmiFeeb3+slayrWknEJoAB9o=";
+    };
+    i686-linux = x86_64-linux;
+    x86_64-darwin = {
+      urlSuffix = "%5bmac%5d%5b1DFC84A6%5d.dmg";
+      hash = "sha256-Sc5BAlpJsffjcNrZ8+VU3n7G10DoqDKQn/leHDW32Y8=";
+    };
+  }.${stdenv.hostPlatform.system} or (throw "Don't know how to fetch source for ${stdenv.hostPlatform.system}!");
+in
+stdenv.mkDerivation rec {
+  pname = "katawa-shoujo";
+  version = "1.3.1";
+
+  src = fetchurl {
+    url = "http://dl.katawa-shoujo.com/gold_${version}/%5b4ls%5d_katawa_shoujo_${version}-${srcDetails.urlSuffix}";
+    inherit (srcDetails) hash;
+  };
+
+  # fetchzip requires a custom unpackPhase to handle dmg, fetchurl cannot handle undmg producing >1 directory without this
+  sourceRoot = ".";
+
+  nativeBuildInputs = lib.optionals stdenv.hostPlatform.isLinux [
+    autoPatchelfHook
+  ] ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    makeWrapper
+    undmg
+  ];
+
+  buildInputs = [
+    freetype
+    zlib
+  ] ++ lib.optionals devendorImageLibs [
+    libjpeg
+    libpng12
+  ] ++ (if withSDL2 then [
+    SDL_compat
+  ] else [
+    SDL
+    SDL_image
+    SDL_ttf
+  ]) ++ lib.optionals stdenv.hostPlatform.isLinux [
+    libX11
+    libXext
+    libXi
+    libXmu
+    libGL
+    libGLU
+  ];
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = let
+    gameArch = "${if stdenv.hostPlatform.isDarwin then "darwin" else "linux"}-${if stdenv.hostPlatform.isx86_64 then "x86_64" else "i686"}";
+    gameSource = if stdenv.hostPlatform.isDarwin then
+      "'Katawa Shoujo'.app"
+    else
+      "'Katawa Shoujo'-${version}-linux";
+    gameInstallDir = if stdenv.hostPlatform.isDarwin then
+      "$out/Applications/'Katawa Shoujo'.app"
+    else
+      "$out/share/katawa-shoujo";
+    gameBin = if stdenv.hostPlatform.isDarwin then
+      "${gameInstallDir}/Contents/MacOS/'Katawa Shoujo'"
+    else
+      "${gameInstallDir}/'Katawa Shoujo'.sh";
+    gameDataDir = if stdenv.hostPlatform.isDarwin then
+      "${gameInstallDir}/Contents/Resources/autorun"
+    else
+      gameInstallDir;
+    gameLibDir = "${gameDataDir}/lib/${gameArch}";
+  in ''
+    runHook preInstall
+
+    mkdir -p "$(dirname ${gameInstallDir})"
+    cp -R ${gameSource} ${gameInstallDir}
+
+    # Simplify launcher script
+    cat <<EOF >${gameBin}
+    #!${runtimeShell}
+    exec \$RENPY_GDB ${gameLibDir}/'Katawa Shoujo' \$RENPY_PYARGS -EO ${gameDataDir}/'Katawa Shoujo'.py "\$@"
+    EOF
+
+  '' + (lib.optionalString stdenv.hostPlatform.isDarwin ''
+    # No autoPatchelfHook on Darwin
+    wrapProgram ${gameBin} \
+      --prefix DYLD_LIBRARY_PATH : ${lib.makeLibraryPath buildInputs}
+  '') + ''
```

```suggestion
  '' + lib.optionalString stdenv.hostPlatform.isDarwin ''
    # No autoPatchelfHook on Darwin
    wrapProgram ${gameBin} \
      --prefix DYLD_LIBRARY_PATH : ${lib.makeLibraryPath buildInputs}
  '' + ''
```
those are optional

also using buildInputs in a wrapper like this is not that nice but works